sycl/source/detail/program_manager/program_manager.hpp

```diff
@@ -359,8 +360,16 @@ class ProgramManager {
   ~ProgramManager() = default;
 
   template <typename NameT>
-  bool kernelUsesAssert(const NameT &KernelName) const {
-    return m_KernelUsesAssert.find(KernelName) != m_KernelUsesAssert.end();
+  bool kernelUsesAssert(const NameT &KernelName,
+                        KernelNameBasedCacheT *KernelNameBasedCachePtr) const {
+    if (!KernelNameBasedCachePtr)
+      return m_KernelUsesAssert.find(KernelName) != m_KernelUsesAssert.end();
```

Comment:
What conditions could lead to us having a kernel name but no cache associated with it?

Reply:
An application that hasn't been recompiled since the cache was added.

Reply:
Sure, I think it makes sense. Since this is also applicable to the UR kernel cache logic introduced by the original cache PR, I'd rather apply this to both in a separate PR.